src/shared/HtmlReady.js

```diff
@@ -262,18 +262,24 @@ function linkify(content, mutate, hashtags, usertags, images, links) {
     });
 
     // usertag (mention)
-    content = content.replace(/(^|\s)(@[a-z][-\.a-z\d]+[a-z\d])/gi, user => {
-        const space = /^\s/.test(user) ? user[0] : '';
-        const user2 = user.trim().substring(1);
-        const userLower = user2.toLowerCase();
-        const valid = validate_account_name(userLower) == null;
-        if (valid && usertags) usertags.add(userLower);
-        if (!mutate) return user;
-        return (
-            space +
-            (valid ? `<a href="/@${userLower}">@${user2}</a>` : '@' + user2)
-        );
-    });
+    // Cribbed from https://github.com/twitter/twitter-text/blob/master/js/twitter-text.js#L90
+    content = content.replace(
+        /(^|[^a-zA-Z0-9_!#$%&*@＠\/]|(^|[^a-zA-Z0-9_+~.-\/]))[@＠]([a-z][-\.a-z\d]+[a-z\d])/gi,
```

Comment:
This is getting pretty hairy. Why is the middle block `(^|[^a-zA-Z0-9_+~.-\/])` needed? Can it be merged with the first?

Also, it looks like that expression will match `[@s0u1` and `@@s0u1` is that what we want? What if the user wrote` [@lbah has a bad website](https://badwebsite.com)`
@roadscape @bnchdrff

Reply:
looks like twitter just rewrote their library, which i was referencing. i fixed the link to point at the specific revision: https://github.com/twitter/twitter-text/blob/v1.14.7/js/twitter-text.js#L90

yes it's complicated, because what we're doing is complicated ;) yes we want to match both of those things. markdown links will have already been converted to HTML by the time it gets here. i added a test to cover links with mentions in them.